### ISSUE TITLE
Replace all instances of "report" in Publisher

### ISSUE THREAD

**Issue by hobuobi** [Recidiviz/justice-counts#165]
**What needs to be done? Why does it need to be done?**
**tl;dr:** The Bureau of Justice Administration insists that agencies are not "reporting" and we are not "collecting" data (ping for details). **As such, we need to remove all instances of the word “report” from Publisher.**  

**Additional context**
All instances of the word Report have been documented [here (in Notion)](https://www.notion.so/recidiviz/Changing-Instances-of-Report-298ccb3cdc3844868f60cfd79bc38967), along with proposed alternatives. General rules for the name changes: 
- Where possible, "report" (used as a verb) should be replaced with "share" or "enter". 
- Reports will be renamed to Records (also taking suggestions for alternative names if this doesn't feel right). 
    - Variable names, functions, databases, etc could still reference "Reports"; will leave up to the engineering team as to what to do about those.

Please look out for any other instances of the word "report" that I might have missed! Ideally this is completed by **Tuesday, November 22** to help unblock conversations with BJA.
